### ISSUE TITLE
Multi-class classification

### ISSUE THREAD

**Issue by Pawelbure** [pytorch/captum#355]
Is there an example for using captum with multi-class classification (similar to your tutorial with the IMDB dataset with binary classification)?

Background: I used a BERT model for multi-class classification (3-way), and 2 classes get positive attribution values whereas the other class gets negative attribution values. I am wondering why the one class gets negative attribution values, while the predictive performance for this class is also good (F1 approx. .75).

This is the model I was using:

```
class BertMultiClassifier(nn.Module):
    def __init__(self, bert_model_path, labels_count, hidden_dim=768, dropout=0.1):
        super().__init__()

        self.config = {
            'bert_model_path': bert_model_path,
            'labels_count': labels_count,
            'hidden_dim': hidden_dim,
            'dropout': dropout,
        }

        self.bert = BertModel.from_pretrained(bert_model_path)
        self.dropout = nn.Dropout(dropout)
        self.linear = nn.Linear(hidden_dim, labels_count)
        self.sigmoid = nn.Sigmoid()

    def forward(self, tokens, masks):
        _, pooled_output = self.bert(tokens, attention_mask=masks, output_all_encoded_layers=False)
        dropout_output = self.dropout(pooled_output)

        linear_output = self.linear(dropout_output)
        proba = self.sigmoid(linear_output)

        return proba
```

And this is the captum code:
```
def predict(inputs, attention_mask=None):
    return model(inputs, masks=attention_mask, )

scores = predict(input_ids, attention_mask=mask )
lig = LayerIntegratedGradients(model, model.bert.embeddings)

attributions, delta = lig.attribute(inputs=input_ids,
                                    baselines=ref_input_ids,
                                    additional_forward_args=( mask ),
                                    target=0,
                                    return_convergence_delta=True,
                                    n_steps = 50)
```

**Comment by NarineK** [comment]
Hi @Pawelbure !
Thank you for the question. Currently we do not attribute to multiple classes at the same time, partly also for the performance reasons. But you can attribute to each class independently and compare the results.
Another way of looking into the combined contribution is to combine those classes, linearly or other ways, and attribute to the output.
To make the attribution more meaningful you might want to choose more meaningful baseline or multiple of them and average across them. Convergence delta can be also a useful hint about the quality of the attributions.

**Comment by Pawelbure** [comment]
Hi @NarineK ,
thanks so much for your reply! The idea of multiple baselines seems very interesting to me, also the analysis of delta. I will try these.

Could you elaborate what you mean by "attribute to each class independently"? 

What I would try is make a similar analysis as your IMDB model for binary sentiment classification. I could separate a 3-way model into 3 2-way models and get the attributions for each class. Is that what you meant?

**Comment by NarineK** [comment]
Hi @Pawelbure, 

With "attribute to each class independently" I meant that if you had 5 classes output and would like to attribute for example to two of them simultaneously you'd need to attribute first to one class with `attribute(input, target=1)` then to the other one with `attribute(input, target=2) `. So you have to call attribute twice. 

Your question looks a bit different. Sorry, I miss-interpreted it. As I understand your problem is that the prediction accuracy is high but the attribution score is low for a particular class in a 3-class classification problem. Is that right ?

I see that you specified `target=0`. Is `linear_output` output-ing the logits for 3 classes ? Also, I'd actually avoid the last step `proba = self.sigmoid(linear_output)` and return `linear_output`. Sigmoid might lead to very small probability scores and it's better to attribute the logits to the inputs.
1. You can have a model that outputs 3 classes and attribute to each of them one at a time using `target` attribute and corresponding target index.
2. Your second suggestion about 3 2-way models  sounds interesting but either way you should be able to see meaningful attributions. If attributions aren't meaningful then we need to look into individual cases and the model.
Hope this helps. Let me know.

**Comment by Pawelbure** [comment]
Hi @NarineK ,
thanks again for clarification, your detailed description and your suggestions. These sound very promising to me and I can proceed with your input!
